### ISSUE TITLE
Lazy-load Async Component shouldn't be loaded with website loaded

### ISSUE THREAD

**Issue by hereiscasio** [nuxt/nuxt.js#9457]
### ◎ Versions

- nuxt: v2.15.3 ( Using latest create-nuxt-app )
- node: v14.15.4

### ◎ Note：Abbreviation
LLAC = Lazy-Load Async Component

### ◎ Reproduction
For Demo Repo, see [this](https://github.com/hereiscasio/Lazy-load-Async-Component) ( note, plz check the home page/route )

Before we dig into demo repo, I want to pick up the key part of the code as shown below
```vue
<template>
...
    <v-sheet v-intersect="onIntersect">
        <LazyHydrate :trigger-hydration="isIntersecting" >
             <chip-async v-if="isIntersecting"></chip-async>
        </LazyHydrate>
    </v-sheet>
...
<script>
   export default {
      data: () => {
        return {
          isIntersecting: false,  
        }
      },     
      components: {
        'chip-async': () => import('~/components/chip-async/index.vue'),
...
```
The _**chip-async**_ will only be loaded when I make _**v-sheet**_ visible ( i.e. by using scrolling ) , 
so in this case, _**chip-async**_ is so called "LLAC"
it works perfectly！
( i think we can ignoring the implementation of `onIntersect` )

BUT！
If I set `isIntersecting: process.client ? false : true`, 
then  _**chip-async**_ will be loaded automatically when you refresh the page ( without doing any scrolling )

**I HAVE NO IDEA WHY IS THAT❓**

Note that, for why i need to use `process.client`, see below, "Why I want to do this？" 

<details open>
<summary>Why I want to do this？</summary>
<br/>
TL;DR<br/> 
👉   I need to take care of both  A) Website Performance  B) SEO
<br/><br/>
I want to introduce LLAC concept into our website homepage
<br/>
But it is harmful for SEO  <br/><br/>
i.e. 
when i viewed index.html page source, LLAC will not be rendered in server side<br/>
<br/>
<br/>
So I took a lots of time to survey the possible solutions  <br/>  

Finally, I believe the more elegant solution is...<br/>  

- 𝟏) Force to render "lazy-load async component" in server side
   -  ( by using the nuxt helper : `process.client`. For more info, see demo repo )
   - so that the Google crawler can see the rendered markup
- 𝟐) And lazy hydrate "lazy-load async component" in client side ( by using the plugin of [vue-lazy-hydration](https://github.com/maoberlehner/vue-lazy-hydration) ) when that component is visible in the viewport
   - otherwise the hydration error will be thrown due to the DOM generated from server side & client side mismatch

I think this is doable. <br/>
BUT, before I implement 𝟐), I encountered the issue which I described in Demo repo<br/>

Note that,
- Our website is built by using Nuxt SSR
- If it is possible, hope someone can suggest the much better ( i.e. more elegant ) solution 
   - because I can't find the solution on the internet which can handle SEO while when using LLAC
</details>

### ◎ Steps to reproduce
- open _**Network**_ panel in DevTools
- Just reload my demo repo
- you can see the file naming "chip-async" is loaded automatically ( but it shouldn't )
![image](https://user-images.githubusercontent.com/44331046/122631808-fd037700-d100-11eb-87fa-2e22326ea661.png)

### ◎ What is Expected?
LLAC should be loaded only when I make the async component appearing in the viewport ( i.e. by using scrolling )

### ◎ What is actually happening?
LLAC will be loaded when the website loaded ( but i haven't make that component appearing in the viewport )

**Comment by manniL** [comment]
IIRC this is caused by Vue "directly. See https://github.com/vuejs/vue/pull/10794

**Comment by hereiscasio** [comment]
@manniL, 

Hi, thx replied！！！
I have also looked into issues/posts which related to what you point out, [#10794](https://github.com/vuejs/vue/pull/10794)

So, do you mean that...

Just because I make `<chip-async v-if="isIntersecting"></chip-async>` runnable in server-side
( i.e.  when i set `isIntersecting: process.client ? false : true` )

And it causes Vue to create the script tag like `<script src="/_nuxt/components/chip-async.js" defer></script>`
( that script tag will be injected into html which is rendered by SSR )

That why I will encounter this issue, right？
If this is true, I think I can close this Issue due to it not directly related to Nuxt

Thx your help again！🙇
